### PR TITLE
CI: cargo clippy

### LINKED ISSUE #30: CI: cargo test
I have quite some errors and warnings, maybe I missed installing something?

## Reproduce
Download this [Dockerfile](https://github.com/ElektraInitiative/opensesame/blob/4edc1a4fc38cfc20251432e7df1fd5014e6430f1/scripts/jenkins/Dockerfile)
```
 curl https://raw.githubusercontent.com/ElektraInitiative/opensesame/4edc1a4fc38cfc20251432e7df1fd5014e6430f1/scripts/jenkins/Dockerfile -o Dockerfile
```

Build and run it
```sh
docker build -t temp_sesam . --build-arg VARIANT=bullseye && docker run -it --rm temp_sesam
```` 

Install git and clone the repo 
```sh
apt-get update && apt-get install -y git && git clone https://github.com/ElektraInitiative/opensesame/ && cd opensesame
```

cargo test
```
cargo test
```

## Result
image

### LINKED ISSUE #35: Cargo test inside jenkins image
Cargo test inside the image used in jenkins would currently fail, due to missing ENV variables.
#25 

Adding this to the Dockerfile or as env should fix it. Testet locally.

```
export PKG_CONFIG_PATH="/usr/lib/arm-linux-gnueabihf/pkgconfig"
export PKG_CONFIG_ALLOW_CROSS="true"
```

### LINKED ISSUE #42: No metadata "mountpoint"
image
It's me again with another error. Not sure if I did something or something happened, so I will ask quick. It's the same setup as in #30, now I get this error when I want to run `postinst`.

### PR DESCRIPTION
**Pull request by 4ydan**
#36

closes #35
closes #30
closes #42

### CODE REVIEW

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
@4ydan please also add minimal documentation of how the CI job is configured here, especially the rebuild phrase `jenkins build please`

File: scripts/jenkins/Dockerfile [hunk 1]
```diff
@@ -1,58 +1,49 @@
-# docker build -t opensesame_cross . --build-arg VARIANT=bullseye
+# docker build -t opensesame .
+# docker build -t opensesame . --build-arg VARIANT=buster
+# docker build -t opensesame . --build-arg VARIANT=buster --build-arg RUST_VERSION=1.71
 ARG VARIANT=bullseye
 ARG RUST_VERSION=1.67.0
 FROM rust:${RUST_VERSION}-slim-${VARIANT}
 ENV DEBIAN_FRONTEND=noninteractive
+ARG VARIANT
```
Comment (fel115) [inline scripts/jenkins/Dockerfile:8]: Why do we need this `ARG` declaration twice?

Comment (4ydan) [reply scripts/jenkins/Dockerfile:8]: FROM introduces a new stage and drops all arguments. We can remove the args if u think that we dont need them, I just put them for practice.

**Comment by 4ydan** [comment]
I wish I was already done with this, but I put way more time into it then I first anticipated.

## Dockerfiles and elektra on amd64/armhf
The main issue here is that the elektra installation from max can't be used to also run the tests with cargo.
So I was trying first to create only one dockerfile with many if`s, that does a different installation depending on the buildArgs provided but somehow ended up in creating two separate dockerfiles (which doesn't make me very happy either).

## Finally
The last obstacle now is permissions issues inside the jenkins workspace not allowing me to run `postinst` + additionally #42 

I hope to solve this soon. Any help appreciated even if its just other ideas.

image

**Comment by markus2330** [comment]
> The main issue here is that the elektra installation from max can't be used to also run the tests with cargo.

Why is this so? @atmaxinger is this something known, did you already talk about it?

> creating two separate dockerfiles

I don't think it is a big problem.

> not allowing me to run postinst

See https://www.libelektra.org/devgettingstarted/testing

You need to give write permissions to /etc/kdb and /usr/share/elektra/specification

**Comment by atmaxinger** [comment]
> Why is this so? @atmaxinger is this something known, did you already talk about it?

I guess because when _cross_ compiling, the toolchain will obviously want to execute the tests on the host architecture, i.e. `amd64` and not the target architecture `armhf`. But my packages only provide `armhf` binaries.

**Comment by 4ydan** [comment]
The permissions are fixed now. Only the `no metadata` thing is left now. `postrm` doesnt seem to help here.

**Comment by 4ydan** [comment]
Thx for reminding me today about this one. Ready for merge.

**Comment by markus2330** [comment]
Thx for this super-fast and amazing fix :rocket:

This should finish your affairs with Opensesame for now! :heart:  :hugs:

**Pull request merged**